lib/hawkular/inventory/entities_v4.rb

```diff
@@ -82,6 +82,15 @@ def initialize(hash)
       @_hash = hash.dup
     end
 
+    def children_by_type(type, recursive = false)
+      selected_children = []
+      @children.each do |child|
+        selected_children << child if child.type.id == type
+        selected_children.push(*child.children_by_type(type, recursive)) if recursive
+      end unless @children.nil?
```

Comment:
Maybe the call should fail when children is nil, because it doesn't mean there's no children, it means they haven't been loaded.

Reply:
Makes sense to me.